### PR TITLE
Fix TracerouteResult.world_trace() and set new defaults for matplotlib graphs

### LINKED ISSUE #28: TracerouteResult.world_trace() is broken
TracerouteResult.world_trace() is broken since 2008, according to git blame.

```python
>>> tr, _ = traceroute(["github.com"], minttl=4, maxttl=6)
[..] 
>>> tr.world_trace()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "scapy/layers/inet.py", line 1078, in world_trace
    from modules.geo import locate_ip
ImportError: No module named modules.geo
```

 Fixing the import won't make it work, yet you will open the gate of tricky & historical Scapy internals:
- this code uses specific databases for geographic and IP addresses locations that are either impossible/difficult to build today
- a function is missing in geoip.py. It is likely that it was never imported from Scapy 1
- gnuplot uses a file called world.dat to draw the world map. This file is also hard to retrieve today
- ...

The best solution seems to rewrite it using the geoip module from maxmind, and the matplotlib map extension. However, it introduces two new modules dependencies.

### PR DESCRIPTION
**Pull request by p-l-**
Fix `TracerouteResult.world_trace()` (fixes #28) using Matplotlib and [GeoIP](https://pypi.python.org/pypi/GeoIP/) (instead of the internal geoip sub-module, which has been removed), and set new defaults for Matplotlib graphs (see [updated notebook](https://github.com/p-l-/scapy/blob/fix-geoip/doc/notebooks/graphs-ipids.ipynb)).

### CODE REVIEW

**Comment by guedou** [comment]
Thanks, I will have a look. We definitely need to share our TODOs. I was rewritten this part myself =)

File: scapy/layers/inet.py [hunk 1]
```diff
@@ -1139,19 +1141,42 @@ def world_trace(self):
...
+                line = line.strip()
+                if not line:
+                    print "PLOT"
+                    plt.plot(*zip(*curline), color="r")
+                    curline = []
+                    continue
+                if line.startswith('#'):
+                    continue
+                curline.append(line.split())
+            print "PLOT"
```
Comment (guedou) [inline scapy/layers/inet.py]: ooops =)

File: scapy/layers/inet.py [hunk 2]
```diff
@@ -1139,19 +1141,42 @@ def world_trace(self):
...
+        # Mimic the default gnuplot output
+        if kargs == {}:
+            kargs = MATPLOTLIB_DEFAULT_PLOT_KARGS
+        lines = [plt.plot(*zip(*tr), **kargs) for tr in trt.itervalues()]
+        with open(conf.gnuplot_world) as fdesc:
+            curline = []
+            for line in fdesc:
+                line = line.strip()
+                if not line:
+                    print "PLOT"
```
Comment (guedou) [inline scapy/layers/inet.py]: ooops =)

**Comment by guedou** [comment]
I was working on a similar patch myself. I do agree that the best solution is to remove the legacy Scapy geoip, however I think that the best path is too use http://matplotlib.org/basemap/ in order to definitely remove the dependency on gnuplot and the world.dat file that might be hard to retrieve.

I can provide the additional fix, as my patch uses it.

**Comment by p-l-** [comment]
Thanks @guedou for the careful review! I have removed `print` statements, and rebased against current master.

For your additional fix, that would be great! Do you prefer to merge this one first and then propose another PR, or to create a PR against my branch (p-l-/scapy:fix-geoip) which will update this PR once merged? The former is probably easier, but the latter is fine too.

**Comment by guedou** [comment]
ACK. I will merge this PR.

**Pull request merged**